pkg/querier/tripperware/query_attribute_matcher.go

```diff
@@ -81,6 +92,8 @@ func getOperation(r *http.Request) string {
 		return "labels"
 	case strings.HasSuffix(r.URL.Path, "/values"):
 		return "label_values"
+	case strings.HasSuffix(r.URL.Path, "/metadata"):
```

Comment:
Prometheus has another target metadata API
https://prometheus.io/docs/prometheus/latest/querying/api/#querying-target-metadata. Even though it is not currently supported in Cortex I think it is still better be specific to match the metric metadata API only.

Maybe we can do

```
case strings.HasSuffix(r.URL.Path, "/metadata"):
  if !strings.Contains(r.URL.Path, "target") {
    return "metadata"
  }
  return "other"
```